lib/irclib/irc.js

```diff
@@ -185,13 +186,15 @@ var getBotClient = function(server) {
 };
 
 module.exports.connect = function() {
-    var defer = q.defer();
+    var defer = promiseutil.defer();
     store.waitForDatabase().done(function() {
         var promises = [];
         servers.forEach(function(server) {
             promises.push(loginToServer(server));
         });
-        q.allSettled(promises).then(function() {
+        Promise.all(promises).then(function() {
```

Comment:
`allSettled`
Returns a promise that is fulfilled with an array of promise state snapshots, but only **after all the original promises have settled**, i.e. become either fulfilled **or rejected**.

`Promise.all`
Given an Iterable(arrays are Iterable), or a promise of an Iterable, which produces promises (or a mix of promises and values), iterate over all the values in the Iterable into an array and return a promise that is fulfilled when all the items in the array are fulfilled. The promise's fulfillment value is an array with fulfillment values at respective positions to the original array. **If any promise in the array rejects**, the returned promise is rejected with the rejection reason

So AFAIK, the promise will resolve early with bluebird the moment any one of the promises rejects, but this doesn't happen with Q.